phylogeny/prototypeSelection.py

```diff
@@ -511,7 +519,9 @@ def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
     # the element to remove first is the one that has smallest distance to all
     # other. "Removing" works by tagging its distance-sum as infinity. Plus, we
     # decrease the number of available elements by one.
-    minElmIdx = currDists.argmin()
+    minElmIdx = next(x for x in currDists.argsort()
```

Comment:
I think the slow down comes from the fact that you're sorting, specially in line 534, that the sorting is inside a for loop (which means that you're sorting multiple times). Instead of argsort use [argmin](https://docs.scipy.org/doc/numpy/reference/generated/numpy.argmin.html) since you're only interested in the minimum one.

The for loop is also performing a filtering (due to the include parameter). Instead of filtering constantly the entire dataset, perform a first pass on the data to filter the data and just work on the filtered data.